### PR TITLE
Decision Record Template

### PR DESCRIPTION
**Pull request by dpalomino**
PR for DR Template

### CODE REVIEW

**Review by oliverroick — CHANGES_REQUESTED** [review]
Thanks for getting this started! 

Some thoughts:

- I think, we need some documentation about the whole process itself. It's not 100% clear to me at this point what documents (I'm thinking of the Github ticket, the decision record, wireframes, functional spec) we're creating and where they live. We need to update the [Development process guide wiki page](https://github.com/Cadasta/cadasta-platform/wiki/Development-process-guide). 
- I think we can add a bit more structure. We have two high-level sections: Functional Spec and then Technical spec. All other sections should be one level below.
- I don't understand the difference between "Description of the implemented use cases" and the information that is already being provided in the GitHub ticket. Is it the same; do we just copy the information from the GitHub ticket?
- Should we have a template for the definition of done? A bunch of boxes that always need to be ticket?

**Comment by dpalomino** [comment]
Thanks a lot @oliverroick for the feedback!

> I think, we need some documentation about the whole process itself. It's not 100% clear to me at this point what documents (I'm thinking of the Github ticket, the decision record, wireframes, functional spec) we're creating and where they live. We need to update the Development process guide wiki page.

Once the DR is approved I can update the wiki, thanks for the heads-up. 

> I think we can add a bit more structure. We have two high-level sections: Functional Spec and then Technical spec. All other sections should be one level below.

Good point. Added in the doc. 

> I don't understand the difference between "Description of the implemented use cases" and the information that is already being provided in the GitHub ticket. Is it the same; do we just copy the information from the GitHub ticket?

It was originally intended to provide a more detailed general explanation about the implementation, but thinking again on this I guess this could be also handled in the "Overview of the suggested solution", so maybe is easier to remove that former one. 

> Should we have a template for the definition of done? A bunch of boxes that always need to be ticket?

Well, I'm not sure if we are going to be able to have a common template for this, as the definition of done could be very different depending on each feature. I was thinking that the definition of done could be a first version of the test case to be included in the Manual test cases spreadsheet. 

Something similar to we were using in the original backlog spreadsheet, like for example (for attaching resources to entities via xforms): 

1. Within a xform, include an image, audio and video resource in location_attributes, party_attributes_default, party_attributes_individual and party_attributes_group
2. Collect the information using geoODK and make sure the resources are included
3. Check that all resources have been properly attached to each entity

What do you think?

**Comment by bjohare** [comment]
It could probably use an **Introduction** section to set context etc. For example with the permissions DR we should explain the reasons for a refactor in the introduction. This could be as simple as a restatement of the problem/context from the Backlog issue.

**Review by oliverroick — APPROVED** [review]

**Comment by dpalomino** [comment]
hi @bjohare !
> It could probably use an Introduction section to set context etc. For example with the permissions DR we should explain the reasons for a refactor in the introduction. This could be as simple as a restatement of the problem/context from the Backlog issue.

I think maybe we can handle this in the **Functional Specification** section. In the backlog issue we should have already included the **Problem Statement** (that's the general description of the need trying to cover) and the **Context/Use Case** (typically providing some real examples of these needs). If we need to add more context or overview prior to the **Technical Specification** I think we can use just the **Functional Specification** section in this DR... does it make sense?

**Review by oliverroick — CHANGES_REQUESTED** [review]
I was about to merge this PR but there are two files now for the DR template this should be cleaned up.

**Comment by dpalomino** [comment]
Thanks @oliverroick, I didn't notice. I've deleted the initial version that was duplicated, it should be ok now...

File: DR_Template.md [hunk 1]
```diff
@@ -18,65 +18,62 @@ In the GitHub issue the following information should have been provided before t
...
 
 
-## Pre-requisites
+### Pre-requisites
 
 Are there any tasks that should be implemented before? If so, please provide links to appropriate DRs and/or GitHub issues. 
 
 
-## Defintion of Done
+### Defintion of Done
```
Comment (amplifi) [inline DR_Template.md]: Typo: should be 'Definition'

File: DR_Template.md [hunk 2]
```diff
@@ -18,65 +18,62 @@ In the GitHub issue the following information should have been provided before t
...
-## Maintainability 
+### Maintainability 
 
 Include here all maintainability aspects that need to be considered. 
 
 
-## Costs considerations
+### Costs considerations
 
 If the suggested solutions has some costs implications. For instance, AWS costs, licensing, etc.
```
Comment (amplifi) [inline DR_Template.md]: Typo: should be 'solution'

Comment (amplifi) [inline DR_Template.md]: Typo: should be 'solution'

**Review by amplifi — APPROVED** [review]
Aside from two small typos, LGTM

**Review by oliverroick — APPROVED** [review]

**Pull request merged**